### PR TITLE
fix: upgrade dependencies and migrate to the new twitch-rs

### PR DESCRIPTION
**Pull request by optimalstrategy**
See individual commits for the changes

### CODE REVIEW

File: scs-db/src/tokens.rs [hunk 1]
```diff
@@ -1,83 +1,81 @@
...
+
+pub async fn create(
+  executor: impl sqlx::PgExecutor<'_> + Copy,
+  // We require `user_id` instead of `username` because the user must already exist
+  user_id: i32,
+  scs_user_api_token: &str,
+  twitch_access_token: &str,
+  twitch_refresh_token: &str,
+) -> Result<Token> {
+  log::info!(
```
Comment (jprochazk) [inline scs-db/src/tokens.rs:20]: :Clueless: logging tokens

File: scs-user-api/src/auth.rs [hunk 2]
```diff
@@ -1,161 +1,161 @@
...
+pub async fn create_token(
+  db: web::Data<db::Database>,
+  client: web::Data<reqwest::Client>,
+  secret: web::Data<ClientSecret>,
+  query: web::Query<TokenQuery>,
+) -> Result<impl Responder> {
+  use crate::ex::*;
+
+  // request authorization
+  log::info!("[authorization] {} {} {}", secret.0, query.code, query.redirect_uri);
```
Comment (jprochazk) [inline scs-user-api/src/auth.rs:30]: 🔢

**Review by jprochazk — APPROVED** [review]
ship it 🚀

File: scs-user-api/src/auth.rs (hunk 2 above)
Comment (optimalstrategy) [reply scs-user-api/src/auth.rs:30]: Should I take out all of those debug logs?

**Pull request merged**